python/ray/train/_checkpoint.py

```diff
@@ -26,8 +26,25 @@
 _CHECKPOINT_TEMP_DIR_PREFIX = "checkpoint_tmp_"
 
 
+class _CheckpointMetaClass(type):
+    def __getattr__(self, item):
+        try:
+            return super().__getattribute__(item)
+        except AttributeError as exc:
+            if item in [
+                "from_dict",
+                "to_dict",
+                "from_bytes",
+                "to_bytes",
+                "from_uri",
+                "to_uri",
```

Comment:
Can we also include: `uri`, `get_internal_representation`, `get_preprocessor` and `set_preprocessor`

Reply:
`uri`, `from_uri` and `to_uri` should have a different message:

To create a checkpoint from remote storage, create a Checkpoint using its constructor instead of `from_directory`. Ex: `Checkpoint(path="s3://a/b/c")`. Then, access the contents of the checkpoint with `as_directory` / `to_directory`. The `ray.train.Checkpoint` no longer provides an API to upload to remote storage using a checkpoint -- use the 
`pyarrow.fs.FileSystem` attribute (`checkpoint.filesystem`) or the  client of your choice to do this instead.

Reply:
`get/set_preprocessor` should have a different error message:

To include preprocessor information in checkpoints, pass it as `metadata` in the `<Framework>Trainer` constructor. Ex: `TorchTrainer(..., metadata={...})`.  After training, access it in the checkpoint via `checkpoint.get_metadata()`.

Reply:
get internal representation can keep the same error message as the current ones

Reply:
Good suggestion, updated!